### ISSUE TITLE
rate-limit-by-key policy no longer works in 2.1.5

### ISSUE THREAD

**Issue by Andrei-Predoiu** [Azure/api-management-self-hosted-gateway#165]
### Report

Rate limiting completely broken on 2.1.5

### Expected Behavior

Rate limiting works as normal

### Actual Behavior

Every API call generates the error below.

### Steps to Reproduce the Problem

1. Deploy the self hosted gateway version 2.1.5
2. Set the following log processing rule on an api
`<rate-limit-by-key calls="200" renewal-period="60" counter-key="@(context.Subscription?.Key ?? "anonymous")" />`
4. Try to call an api with rate limiting enabled

### Logs from self-hosted gateway

```
{"level":"Error","timestamp":"2022-10-17T07:57:11.9366843Z","eventType":"PolicyParsingFailed","message":"key: scope: Api, id: Product-Distribution-Services-API;rev=1, revision: 18192","exception":"System.InvalidOperationException: There is an error in XML document (0, 0).\n ---> System.InvalidOperationException: <rate-limit-by-key xmlns=''> was not expected.\n   at Microsoft.Xml.Serialization.GeneratedAssembly.XmlSerializationReaderRateLimit.Read6_ratelimit()\n   --- End of inner exception stack trace ---\n   at System.Xml.Serialization.XmlSerializer.Deserialize(XmlReader xmlReader, String encodingStyle, XmlDeserializationEvents events)\n   at System.Xml.Serialization.XmlSerializer.Deserialize(XmlReader xmlReader)\n   at Microsoft.WindowsAzure.ApiManagement.Proxy.Runtime.Configuration.PolicyConfigurationParser.ParseConfiguration(Type configurationType, XElement configElement) in D:\\a\\1\\s\\Proxy\\Gateway.Policies\\PolicyConfigurationParser.cs:line 155\n   at Microsoft.WindowsAzure.ApiManagement.Proxy.Runtime.Configuration.PolicyConfigurationParser.ParseSteps(PolicyScopeKey scopeKey, IEnumerable`1 stepConfigurationElements) in D:\\a\\1\\s\\Proxy\\Gateway.Policies\\PolicyConfigurationParser.cs:line 73\n   at Microsoft.WindowsAzure.ApiManagement.Proxy.Runtime.Configuration.PolicyMapBuilder.ParsePolicy(PolicyScopeKey key, XElement root) in D:\\a\\1\\s\\Proxy\\Gateway.Policies\\PolicyMapBuilder.cs:line 211\n   at Microsoft.WindowsAzure.ApiManagement.Proxy.Runtime.Configuration.PolicyMapBuilder.Parse(PolicyScopeKey key, PolicyDescription policy) in D:\\a\\1\\s\\Proxy\\Gateway.Policies\\PolicyMapBuilder.cs:line 187\n   at Microsoft.WindowsAzure.ApiManagement.Proxy.Runtime.Configuration.PolicyMapBuilder.ProcessQueueAsync(CancellationToken cancellation) in D:\\a\\1\\s\\Proxy\\Gateway.Policies\\PolicyMapBuilder.cs:line 158","source":"PolicyMapBuilder"}
{"level":"Error","timestamp":"2022-10-17T07:57:11.9375475Z","eventType":"FailedToProcessRequest","apiId":"Product-Distribution-Services-API","operationId":"get-style-style-number","subscriptionId":"631b156f9d679a8b3da1f41e","tags":16,"httpMethod":"GET","source":"policy-configuration","exception":"Microsoft.WindowsAzure.ApiManagement.Proxy.Runtime.Configuration.PolicyParsingException: Policy parsing failed for Api Product-Distribution-Services-API;rev=1 (revision 18192)\n ---> System.InvalidOperationException: There is an error in XML document (0, 0).\n ---> System.InvalidOperationException: <rate-limit-by-key xmlns=''> was not expected.\n   at Microsoft.Xml.Serialization.GeneratedAssembly.XmlSerializationReaderRateLimit.Read6_ratelimit()\n   --- End of inner exception stack trace ---\n   at System.Xml.Serialization.XmlSerializer.Deserialize(XmlReader xmlReader, String encodingStyle, XmlDeserializationEvents events)\n   at System.Xml.Serialization.XmlSerializer.Deserialize(XmlReader xmlReader)\n   at Microsoft.WindowsAzure.ApiManagement.Proxy.Runtime.Configuration.PolicyConfigurationParser.ParseConfiguration(Type configurationType, XElement configElement) in D:\\a\\1\\s\\Proxy\\Gateway.Policies\\PolicyConfigurationParser.cs:line 155\n   at Microsoft.WindowsAzure.ApiManagement.Proxy.Runtime.Configuration.PolicyConfigurationParser.ParseSteps(PolicyScopeKey scopeKey, IEnumerable`1 stepConfigurationElements) in D:\\a\\1\\s\\Proxy\\Gateway.Policies\\PolicyConfigurationParser.cs:line 73\n   at Microsoft.WindowsAzure.ApiManagement.Proxy.Runtime.Configuration.PolicyMapBuilder.ParsePolicy(PolicyScopeKey key, XElement root) in D:\\a\\1\\s\\Proxy\\Gateway.Policies\\PolicyMapBuilder.cs:line 211\n   at Microsoft.WindowsAzure.ApiManagement.Proxy.Runtime.Configuration.PolicyMapBuilder.Parse(PolicyScopeKey key, PolicyDescription policy) in D:\\a\\1\\s\\Proxy\\Gateway.Policies\\PolicyMapBuilder.cs:line 187\n   at Microsoft.WindowsAzure.ApiManagement.Proxy.Runtime.Configuration.PolicyMapBuilder.ProcessQueueAsync(CancellationToken cancellation) in D:\\a\\1\\s\\Proxy\\Gateway.Policies\\PolicyMapBuilder.cs:line 158\n   --- End of inner exception stack trace ---\n   at Microsoft.WindowsAzure.ApiManagement.Proxy.Gateway.Extensions.PolicyMapHelper.TryAppend(List`1 list, IPolicyEntity policy, IPolicyMapBuilder builder, CancellationToken cancellation) in D:\\a\\1\\s\\Proxy\\Gateway.Policies\\PolicyMapHelper.cs:line 39\n   at Microsoft.WindowsAzure.ApiManagement.Proxy.Gateway.Extensions.PolicyMapHelper.ResolveAsync(IPublisher publisher, IProduct product, IApi api, IOperation operation, IPolicyMapBuilder builder, CancellationToken cancellation) in D:\\a\\1\\s\\Proxy\\Gateway.Policies\\PolicyMapHelper.cs:line 29\n   at Microsoft.WindowsAzure.ApiManagement.Proxy.Gateway.Handlers.PolicySchedulingHandler.ProcessAsync(IPipelineContext context, CancellationToken cancellation) in D:\\a\\1\\s\\Proxy\\Gateway.Policies\\PolicySchedulingHandler.cs:line 40\n   at Microsoft.WindowsAzure.ApiManagement.Proxy.Gateway.Policies.PipelineWalker.ExecuteAsync(IPipelineContext context, IEnumerable`1 steps, CancellationToken cancellation) in D:\\a\\1\\s\\Proxy\\Gateway.Pipeline\\PipelineWalker.cs:line 49\n   at Microsoft.WindowsAzure.ApiManagement.Proxy.Gateway.PipelineExecutor.ExecuteAsync(IPipelineContext context, CancellationToken cancellationToken) in D:\\a\\1\\s\\Proxy\\Gateway.Pipeline\\PipelineExecutor.cs:line 204","transportError":0,"httpError":0}
```

### Self-hosted Gateway Version

2.1.5

### Deployment

Self-Managed (YAML/Helm)

### Platform

Google Cloud

### Kubernetes Version

1.22

**Comment by tomkerkhove** [comment]
Can you elaborate on what version you were previously on and if it worked there?

I will try to reproduce it.

**Comment by tomkerkhove** [comment]
I can reproduce this and will work on a fix + see why our automated testing didn't catch this.

**Comment by Andrei-Predoiu** [comment]
It was working on 2.1.5, not sure about 2.1.4

**Comment by tomkerkhove** [comment]
This is now fixed in v2.1.6 - Thanks for reporting!
https://github.com/Azure/api-management-self-hosted-gateway/releases/tag/Container-v2.1.6
